### ISSUE TITLE
Error when loading results in compare view

### ISSUE THREAD

**Issue by lewfish** [project-icp/bee-pollinator-app#115]
If you trigger an initial model run and then switch to the compare view before the results have returned, the following error message appears in the console, the spinner keeps going, and the new results never appear. This is true even after the results being returned by the backend. This also happens when triggering a second model run in the modeling view, and then switching to the compare view, although the error in the console doesn't appear.

```
views.js:689 Uncaught TypeError: Cannot read property 'show' of undefined
    at child.showDetailsRegion (http://localhost:8000/static/js/main.js:8551:32)
    at Array.<anonymous> (http://localhost:8000/static/js/main.js:8533:18)
    at fire (http://localhost:8000/static/js/vendor.js:42654:30)
    at Object.fireWith [as resolveWith] (http://localhost:8000/static/js/vendor.js:42766:7)
    at Array.<anonymous> (http://localhost:8000/static/js/vendor.js:42894:16)
    at fire (http://localhost:8000/static/js/vendor.js:42654:30)
    at Object.fireWith [as resolveWith] (http://localhost:8000/static/js/vendor.js:42766:7)
    at Array.deferred.(anonymous function) (http://localhost:8000/static/js/vendor.js:42856:34)
    at fire (http://localhost:8000/static/js/vendor.js:42654:30)
    at Object.fireWith [as resolveWith] (http://localhost:8000/static/js/vendor.js:42766:7)
```

**Comment by lewfish** [comment]
This doesn't happen in MMW.

**Comment by rajadain** [comment]
Here's what I've discovered so far. In the Compare view, even after polling finishes successfully, the actual value of the `results[n].polling` isn't updated:

![image](https://cloud.githubusercontent.com/assets/1430060/22480604/f8cc48ee-e7bf-11e6-964c-dce3dd635330.png)

This is confusing, because `setPolling(false)` executes correctly. This does not happen if we stay on the modeling view. It is possible this is a side effect of #103.

**Comment by rajadain** [comment]
Okay, confirmed that the spinners that don't stop also don't stop in MMW. This seems to be caused by the copy of the project that is made before entering Compare view.

When we started wanting to run models from the Compare view in https://github.com/WikiWatershed/model-my-watershed/pull/673, we wanted to do it in a way that did not affect the state of the Modeling view. Thus, we make a sandboxed copy of the project before handing it to the Compare view.

As a result, if there is a pending operation, thus making `polling: true` in the original project, that same value is passed to the Compare view. However, when the results do come back, it is the original project that is set to `polling: false`, not the sandboxed copy, which keeps the spinners spinning.

If there isn't a use case for running models from the Compare view in this project (and that so far seems to be the case), I can rip out all that code and thus fix this bug.

/cc @ajrobbins @mmcfarland

**Comment by mmcfarland** [comment]
Yeah, I got caught by this too for the `shared_modifications` stuff.  I ended up copying over the things I need to the compare copy. We don't have plans to allow changes to inputs in compare - I held off on removing it because I _could_ see us letting someone change managed hives, but I'd rather not spend time supporting a non-feature at the expense of just removing some non-consequential code (that has a bug).  I say strike it.
